cmd/agent/dogstatsd/global.go

```diff
@@ -0,0 +1,16 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package dogstatsd
+
+import (
+	"github.com/DataDog/datadog-agent/comp/dogstatsd/server"
+)
+
+// DSD is the global dogstatsd instance
+// TODO: (components) This should be removed when all downstream dependencies
+// are migrated to components such that this can be injected instead of a shared
+// global reference.
+var DSD server.Component
```

Comment:
It would be great to remove this variable. I had a quick look and it seems doable. Can you track this task?

Reply:
I was able to remove it in all places except for JMX fetch. Core checks all use `init` hooks so until we refactor that - we have to use a global in that case. 
I updated the comment here to clarify that. Ill open a work item to track this refactor. In the meantime, I've remove the global from all other locations :)